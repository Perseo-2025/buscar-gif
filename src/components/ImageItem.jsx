import PropTypes from "prop-types";

export default function ImageItem({ tags, id, url }) {
    return (
      <div className=" text-white rounded-2xl shadow-md overflow-hidden transition-transform transform hover:scale-105">
        <img
          key={id}
          src={url}
          alt={tags}
          className="w-full h-48 object-cover"
        />
        <div className="p-3 text-center">
          <p className="text-sm text-white">{tags}</p>
        </div>
      </div>
    );
  }
  
  ImageItem.propTypes = {
    tags: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired
  }
  /*
  Tarea
  1.AÃ±adir PropTypes
  a. title obligatoria
  b. url obligatoria
  2. Evaluar el snapshot
  */