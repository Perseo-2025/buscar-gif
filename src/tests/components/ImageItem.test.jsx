import { render } from "@testing-library/react"
import ImageItem from "../../components/ImageItem"

describe('Pruebas en el componente ImageItem', () => {
    const id = 1
    const tags = 'rose'
    const url = 'https://example.com/image.jpg'
    test('debe de hacer match con el snatshop', () => {
        
        const {container} = render(<ImageItem tags={tags} id={id} url={url} />)
        expect(container).toMatchSnapshot()


    })

    test('debe de mostrar la imagen con la url y el alt', () => {
        const { getByAltText } = render(<ImageItem tags={tags} id={id} url={url} />)
        const image = getByAltText(tags)
        expect(image.src).toBe(url)
        expect(image.alt).toBe(tags)
    })

    test('debe de mostrar el titulo en el componente', () => {
        const { getByText } = render(<ImageItem tags={tags} id={id} url={url} />)
        const title = getByText(tags)
        expect(title).toBeTruthy()
    })
})