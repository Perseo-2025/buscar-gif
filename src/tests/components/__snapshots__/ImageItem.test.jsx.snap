// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ImageItem debe de hacer match con el snatshop 1`] = `
<div>
  <div
    class=" text-white rounded-2xl shadow-md overflow-hidden transition-transform transform hover:scale-105"
  >
    <img
      alt="rose"
      class="w-full h-48 object-cover"
      src="https://example.com/image.jpg"
    />
    <div
      class="p-3 text-center"
    >
      <p
        class="text-sm text-white"
      >
        rose
      </p>
    </div>
  </div>
</div>
`;
